src/features/todos.ts

```diff
@@ -1,8 +1,32 @@
-import { createSlice } from '@reduxjs/toolkit';
-import { Todo } from '../types/Todo';
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Todo, TodosState } from '../types/Todo';
+
+const initialState: TodosState = {
+  todos: [],
+  loading: false,
+  error: null,
+};
 
 export const todosSlice = createSlice({
   name: 'todos',
-  initialState: [] as Todo[],
-  reducers: {},
+  initialState,
+  reducers: {
+    fetchTodosStart(state) {
+      state.loading = true;
+      state.error = null;
+    },
+    fetchTodosSuccess(state, action: PayloadAction<Todo[]>) {
+      state.loading = false;
+      state.todos = action.payload;
```

Comment:
Here, you're replacing the entire todos array with the new todos fetched from the server. This is fine if you always want to completely replace the todos in your state whenever new ones are loaded. However, if you want to add features like pagination or incremental loading of todos in the future, you might want to consider merging the new todos with the existing ones instead of replacing them. You could use the spread operator to achieve this, something like `state.todos = [...state.todos, ...action.payload];` This way, the new todos will be added to the end of the existing todos array.